### ISSUE TITLE
Is this evaluation script really correct?

### ISSUE THREAD

**Issue by 4vik** [skrish13/SKU110K-evaluation#5]
The reason I am asking is this:

On the SKU110K dataset, using this script, I can reproduce exactly the same results as in your paper, i.e. an AP of 0.685 (using your predictions.csv file from your Faster-RCNN model). 

However, using your same evaluation script to evaluate the original model (proposed by Goldman et al.) on the SKU110K dataset, I actually get an AP of 0.859, which is very high. It is also much higher than your proposed Faster-RCNN-model's performance, and much higher than what is mentioned in the Goldman paper. Which is very strange, as I am using exactly the same model as Goldman proposed.

This makes me suspect that there is a mistake in this evaluation script (it gives much higher results than what the reality is...) although I can't find it. Any thoughts?

**Comment by 4vik** [comment]
Update: I believe I have found the bug in https://github.com/skrish13/SKU110K-evaluation.

In the file calc_acc.py, the line `entry['bbox'] = [row['x1'], row['y1'], row['x2'], row['y2']]` is almost certainly wrong. If one reads the documentation for the COCO dataset (specifically http://cocodataset.org/#format-results), one can see that the bbox parameters should be "x1, y1, width, height" and not "x1, y1, x2, y2". Thus, the correct line of code there would be: `entry['bbox'] = [row['x1'], row['y1'], row['x2']-row['x1'], row['y2']-row['y1']]`.

After correcting this line (appears twice in the calc_acc.py file), I got a lot more reasonable mAP values.

**TLDR: I believe there is an error in the https://github.com/skrish13/SKU110K-evaluation evaluation script, use it with caution.**

(BTW:
If I am correct, then unfortunately, this also means that all the official results mentioned here are wrong (i.e. the results are worse in reality): https://github.com/ParallelDots/generic-sku-detection-benchmark (I double-checked their results and they are indeed wrong)

**Comment by skrish13** [comment]
Ah! This is great!! Thanks for finding the bug!! I always felt there was something strange, which is why I open-sourced this back in June :). Will update the results in the other repository and the paper accordingly.

I have corrected it now, so closing this issue. Thanks again!

PS: Were you able to replicate the results from the SKU110K paper after correcting this part?

**Comment by 4vik** [comment]
Hey @skrish13 , yes, I was able to replicate them :) Unfortunately, it also meant that the results you published in https://github.com/ParallelDots/generic-sku-detection-benchmark are actually quite much lower... for example, your results on the Grocery_products dataset are in reality an mAP of 0.234 instead of the 0.537 that were reported. Do you agree or am I missing something?

**Comment by skrish13** [comment]
Yup, I have changed the results as well (https://github.com/skrish13/SKU110K-evaluation/blob/master/README.md). I am re-calculating on other datasets and will update them as well.

**Comment by skrish13** [comment]
> Hey @skrish13 , yes, I was able to replicate them :) 

Ah that's great! I'm interested in their mAP at IoU=0.5 which isn't given in the paper. Do you mind sharing the different accuracy metrics which you got while replicating? @4vik

**Comment by 4vik** [comment]
Yeah, sure, I got an mAP of 0.853 at IoU=0.5, using their pretrained model (which they shared in one of their repository issues) for prediction... means their model definitely has room for improvement in the localization

**Comment by eg4000** [comment]
Hi @4vik , @skrish13, 
Thank you very much in your help to validate the evaluation process.

We are actually going to organize a challenge for this task in our workshop in the upcoming CVPR. we will update the project web page with more detail soon.

Regards,
Eran.

**Comment by skrish13** [comment]
Hi @eg4000 ,

Is there a timeline for the workshop details? (Is the due date in March or perhaps later in May?)

**Comment by eg4000** [comment]
@ehudbarnea Can you please update?

**Comment by retailvisionworkshop** [comment]
Hi,

We'll setup the website during February (hopefully as close as possible to the beginning of February). It will include the same dataset and new evaluation code based on Ross Girshick's implementation (this is where Eran took the evaluation used in his paper). Later on (maybe in April/May, we will update) we will release new test data without labels.

(Written by @EhudBarnea, this account will probably be used for the website)

**Comment by skrish13** [comment]
Hi @retailvisionworkshop @EhudBarnea, 

I see that you've updated the website!
Maybe I'm jumping the gun here, but where can we access the AliProducts dataset? :)

**Comment by retailvisionworkshop** [comment]
Thanks for the interest :)
The AliProduct dataset will only be available later on, the current plan is to have it published in 2-3 months.

**Comment by retailvisionworkshop** [comment]
Hi,
Just updating that the workshop page has been updated with the detection challenge info as well as the code we will use for method evaluation (I tried to make as few modifications as possible to the base code). You're welcome to have a look:
https://retailvisionworkshop.github.io/detection_challenge_2020/

**Comment by skrish13** [comment]
Hi @retailvisionworkshop @EhudBarnea,

I was curious if the papers accepted by the workshop would be published in the proceedings/CVF Open Access as not all workshops publish..

[1] http://openaccess.thecvf.com/CVPR2019_workshops/menu.py

**Comment by retailvisionworkshop** [comment]
Only a part of the workshops publish in the proceedings. We have decided not to do so this year, so paper will be present as posters and in the website and some will be chosen to give a talk during the workshop.

**Comment by skrish13** [comment]
Ah okay. Thanks for the quick response :)
